### PR TITLE
Refactoring and tidying

### PR DESCRIPTION
**Pull request by RabeaGleissner**
@jsuchy @felipesere

This PR includes some refactoring based on the discussion we had in my IPM yesterday:
- winner method no longer returns two different types of data
- replaced haml template with erb
- introduced the Marks module for player marks
- removed the inheritance from player

---

The speed test that checks if the computer player makes the first move in under one second failed on Travis (it works fine on my machine). 

For now I've updated the test to assert that the first move is faster than 2 seconds which seems to work for Travis. Is there another way of getting around this? I feel like I'm cheating!

### CODE REVIEW

File: lib/computer_player.rb [hunk 1]
```diff
@@ -46,27 +47,38 @@ def minimax(depth, board, current_mark, alpha, beta)
...
+      depth
     else
-      return - depth
+      -depth
     end
   end
 
   def switch_mark(mark)
-    mark = mark == :X ? :O : :X
+    mark = mark == X ? O : X
```
Comment (felipesere) [inline lib/computer_player.rb:60]: Could this somehow move to the Marks module? It has no dependency on the board.

File: lib/computer_player.rb [hunk 2]
```diff
@@ -46,27 +47,38 @@ def minimax(depth, board, current_mark, alpha, beta)
...
+  def ready?
+    true
   end
 
   private
 
   def score_favourable_for_computer?(current_mark, score, best_score)
-    current_mark == computer_mark && score[0] > best_score ||
-      current_mark != computer_mark && score[0] < best_score
+    if (computer_mark_is?(current_mark))
```
Comment (felipesere) [inline lib/computer_player.rb:70]: The question mark (pun!) breaks the readability... How about simply computer?(current_mark)

Comment (RabeaGleissner) [inline lib/computer_player.rb:70]: OK, will update the method name.

File: lib/computer_player.rb (hunk 1 above)
Comment (RabeaGleissner) [inline lib/computer_player.rb:60]: Good point! Will move it.

**Pull request merged**